### ISSUE TITLE
gist.el doesn't work with 2 factor Auth

### ISSUE THREAD

**Issue by jjasghar** [defunkt/gist.el#113]
One of my organizations forced 2 factor Auth on me, and now trying to run `M-x gist-buffer-private` I get this:

```
Contacting host: api.github.com:443
Opening TLS connection to ‘api.github.com’...
Opening TLS connection with ‘gnutls-cli --x509cafile nil -p 443 api.github.com’...failed
Opening TLS connection with ‘gnutls-cli --x509cafile nil -p 443 api.github.com --protocols ssl3’...failed
Opening TLS connection with ‘openssl s_client -connect api.github.com:443 -no_ssl2 -ign_eof’...done
Opening TLS connection to ‘api.github.com’...done
Decrypting /Users/jjasghar/.authinfo.gpg...done
json-read: JSON readtable error
```

I'm not sure how to debug this farther, but this is the only change that has happened between Github and my Emacs installation recently.

**Comment by zzamboni** [comment]
I was able to use `gist.el` with 2FA enabled by creating a Personal Access Token with "gist" scope, and using it instead of my password, and also when gist.el asks for your "OAuth Token".

Documentation on how to create the token: https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/

You may need to remove the `[github]` section from your `~/.gitconfig` file, and restart Emacs so that gist.el asks for your information again.

**Comment by chrismoos** [comment]
@jjasghar I ran into what looked like a similar issue, but it turned out to be an issue with the `openssl` binary on my machine not supporting TLSv1.2 (GitHub doesn't support old TLS protocols anymore).

If you have a Mac, make sure you have the latest `openssl` installed via Homebrew (`brew upgrade openssl`) and get the binary on your path:
```
ln -s /usr/local/opt/openssl/bin/openssl /usr/local/bin/openssl
```

Then exit emacs, open a new terminal, and try again.

**Comment by nickanderson** [comment]
> I was able to use `gist.el` with 2FA enabled by creating a Personal Access Token with "gist" scope, and using it instead of my password, and also when gist.el asks for your "OAuth Token".
> 
> Documentation on how to create the token: https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/
> 
> You may need to remove the `[github]` section from your `~/.gitconfig` file, and restart Emacs so that gist.el asks for your information again.

Yeah, it's disappointing that I can't seem to leverage authinfo which can be gpg encrypted and where I store other credentials used by forge and magit. I don't want to put a token into my .gitconfig which is plain text.

**Comment by ayman** [comment]
On Mac Big Sur, emacs 27.1, OpenSSL 1.1.1h  22 Sep 2020 (brew installed) nothing will make private gists work or list. Tried the Personal Access Token and made sure openssl was up to date...nada. Not sure if there's some other setting but there's no error in the *Messages* buffer...just it rolls back to showing only the public gists for me.
